pkg/diagnostics/systemd/unit_status.go

```diff
@@ -24,13 +22,10 @@ func (d UnitStatus) Description() string {
 	return "Check status for related systemd units"
 }
 func (d UnitStatus) CanRun() (bool, error) {
-	if runtime.GOOS == "linux" {
-		if _, err := exec.LookPath("systemctl"); err == nil {
-			return true, nil
-		}
+	if IsSystemd() {
+		return true, nil
 	}
-
-	return false, errors.New("systemd is not present on this host")
+	return false, errors.New("systemd is not present/functional on this host")
```

Comment:
@deads2k it results in more details in the "Not gonna run this" message.

https://github.com/openshift/origin/blob/master/pkg/cmd/experimental/diagnostics/diagnostics.go#L235

It has no other effect, so there's no need to pass back an error unless it will be informative to the user. Which it is here.